src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -95,21 +96,41 @@ export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionC
     }
 
     public async deleteTreeItemImpl(): Promise<void> {
-        const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
-        await client.databases.deleteMethod(azureUtils.getResourceGroupFromId(this.fullId), this.parent.name, this.databaseName);
+        if (this.parent.server) {
+            const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
+            await client.databases.deleteMethod(azureUtils.getResourceGroupFromId(this.fullId), this.parent.name, this.databaseName);
+        } else {
+            const username_connString = nonNullProp(this.parent.connectionString, 'username');
+            const password_connString = nonNullProp(this.parent.connectionString, 'password');
+            const host = nonNullProp(this.parent.connectionString, 'hostName');
```

Comment:
Why do you have to call `nonNullProp` one these connection string properties?

Reply:
Actually nonNullProp is not necessary here, I think it is moved code which was not updated.